File: Java/T5/T05_Dialogos/app/src/main/java/com/borja/dialogos/dialogos/DialogoHora.java
package com.borja.dialogos.dialogos;

import android.app.Dialog;
import android.app.TimePickerDialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import java.util.Calendar;

public class DialogoHora extends DialogFragment {

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        return new TimePickerDialog(getActivity(), (TimePickerDialog.OnTimeSetListener) getActivity(),
                Calendar.getInstance().get(Calendar.HOUR_OF_DAY),
                Calendar.getInstance().get(Calendar.MINUTE),
                true);
    }
}

File: Kotlin/T5/codigos2223/T5_Dialogos/app/src/main/java/com/develop/t5_dialogos/dialogs/DialogoHora.kt
package com.develop.t5_dialogos.dialogs

import android.app.Dialog
import android.app.TimePickerDialog
import android.app.TimePickerDialog.OnTimeSetListener
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import com.develop.t5_dialogos.R
import java.util.Calendar

class DialogoHora: DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        val caledario = Calendar.getInstance();
        val hora = caledario.get(Calendar.HOUR_OF_DAY)
        val minutos = caledario.get(Calendar.MINUTE)
        val dialogo = TimePickerDialog(requireContext(), R.style.DialogoColores  , requireContext() as OnTimeSetListener,hora,minutos,true)
        return dialogo
    }
}